pages/api/models/module.ts

```diff
@@ -0,0 +1,40 @@
+import { model, Schema } from "mongoose";
+
+enum StatusType {
+    COMPLETE = "complete",
+    DRAFT = "draft",
+    PUBLISHED = "published",
+}
+
+export interface ModuleInterface {
+    _id: Schema.Types.ObjectId;
+    title: string;
+    tool: Schema.Types.ObjectId;
+    slides: [Schema.Types.ObjectId];
```

Comment:
Since MongoDB doesn't use foreign keys we need to remember to validate that id's like tool id/slide ids are valid before adding them. Can we add a comment for this for the tool and slides columns (so we remember this when implementing endpoints)